rosidl_typesupport_introspection_c/CMakeLists.txt

```diff
@@ -40,6 +40,10 @@ ament_index_register_resource("rosidl_runtime_packages")
 
 if(BUILD_TESTING)
   find_package(ament_lint_auto REQUIRED)
+  find_package(rosidl_typesupport_interface REQUIRED)
+  # Give cppcheck hints about macro definitions coming from outside this package
+  get_target_property(
+    ament_cmake_cppcheck_ADDITIONAL_INCLUDE_DIRS rosidl_typesupport_interface::rosidl_typesupport_interface INTERFACE_INCLUDE_DIRECTORIES)
```

Comment:
note to self, `ament_cmake_cppcheck_ADDITIONAL_INCLUDE_DIRS` comes from here: https://github.com/ament/ament_lint/blob/8bf194aa1ac282db5483dd0d3fefff8f325b0db8/ament_cmake_cppcheck/cmake/ament_cppcheck.cmake#L18-L20